Gateway/Request/RefundRequest.php

```diff
@@ -46,7 +46,9 @@ public function build(array $buildSubject)
         $payment = $paymentDataObject->getPayment();
         $transactionId = $payment->getAdditionalInformation(self::TRANSACTION_ID) ?:
             $payment->getAdditionalInformation(self::CHARGE_ID);
-        $amountInCents = Util::formatToCents($buildSubject['amount']);
+        $countryCode = $payment->getAdditionalInformation(self::COUNTRY_CODE) ?: 'USA';
+        $creditMemoAmount = $payment->getCreditmemo()->getGrandTotal();
+        $amountInCents = Util::formatToCents($creditMemoAmount);
```

Comment:
will this support a partial amount refund?

Reply:
Yes